packages/zoe/src/contracts/auction/index.js

```diff
@@ -102,20 +102,20 @@ const start = zcf => {
     return bidSeats.map(seat => seat.getAmountAllocated('Bid', bidBrand));
   };
 
-  const getSessionDetails = () => {
+  const getSessionDetails = async () => {
     const sellerProposal = sellSeat.getProposal();
     return harden({
```

Comment:
Your use of `await` nested within an expression violates our `await` design rule. This code is a good example of why. Looking at this code, is it clear whether `sellerProposal.want.Ask` and `getCurrentBids()` are evaluated before or after the `await` point?

Instead, if you want to use `await` here, I suggest you rename the variable that might hold a promise for a time authority to something else, and then await it at function top level to obtain the non-promise time authority variable.

```suggestion
    const timeAuthority = await timeAuthorityP;
    return harden({
```